Feature contact validation

eduardosancho opened the pull request:
What was done in this PR:

-Remove 'required' property from form elements.
-Use javascript to validate the information inserted by the user.

- [x] Name field cannot be empty
- [x] Email field cannot be empty
- [x] Email field cannot have capital letters
- [x] Email field must have a valid email address. (following syntax abc@domain.com)
- [x] If a field has errors, a message is displayed near the submit button.

david-lafontant committed:
Create the validation script and link it to index

eduardosancho committed:
Add script to play with form validation

eduardosancho committed:
Add validation to old form, eliminating required

eduardosancho committed:
Add dlink to header collab button

eduardosancho committed:
Add form validation to check for capitals in email

eduardosancho committed:
Fix linter suggestions.

oxenprogrammer commented on index.html:
```diff
@@ -228,22 +228,22 @@ <h1> Get started</h1>
-                        required></textarea>
-                    <button type="submit" id="submit-form">Start collaboration</button>
+                    <textarea id="msg" name="message" placeholder="Enter your message here..." maxlength="500"></textarea>
+                    <button type="submit" class="full" id="submit-form">Start collaboration</button>
```
**Changes Required**
- [x] Please let us also try and validate `textarea` field. Apparently one can submit the form without any message.

oxenprogrammer replied:
-  [x] Also your error message should be closer to the submit button has described in the requirements.

oxenprogrammer reviewed (changes requested):
Hi @eduardosancho, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

eduardosancho committed:
Move form messages closer to submit button

eduardosancho committed:
Add success and error colors to form fields

eduardosancho committed:
Add validation messages to contact form.

-Add validation for textarea
-Give all fields their own validation message.
-If a validation is corrected, message disappers.

eduardosancho committed:
Fix linter suggestions

eduardosancho commented:
It seems linters are not working on GitHub right now.
I ran the checks locally and these were the results, everything passed fine.

![image](https://user-images.githubusercontent.com/53308506/146628974-17a4c582-e1a7-4ed1-ae47-d4eb76f284cc.png)

Cathella reviewed (approved):
## 🟢 Status: Approved 🎉💪🏻

### Hi @team ✋🏻, 

You have done well implementing the requested changes 💪🏻, You form now submits and is well validated, Kudos to you 🎉🍾!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

eduardosancho committed:
Remove html hint report

eduardosancho committed:
Remove hint-report file

eduardosancho committed:
Merge branch 'feature-contact-validation' of github.com:eduardosancho/setup_and_mobile_version_skeleton into feature-contact-validation
Removed hint report files

eduardosancho committed:
Move figma template to acknoledgements section

eduardosancho committed:
Add table to authors section

eduardosancho committed:
Add link to Live Server extension

eduardosancho committed:
Merge branch 'main' into development

eduardosancho committed:
Arrange error messages vertically

eduardosancho committed:
Merge branch 'development' into feature-contact-validation

eduardosancho committed:
Update linters dependencias

eduardosancho committed:
Display validation errors in contact form

eduardosancho committed:
Reduce width of content on large screens

eduardosancho committed:
Add AWS S3 to Reservify

eduardosancho committed:
Readjust projects translation on hover

eduardosancho committed:
Add color transition to cv button

eduardosancho committed:
Invert icon color on button hover

eduardosancho committed:
Run linters
